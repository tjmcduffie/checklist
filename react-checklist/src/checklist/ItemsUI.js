import classnames from 'classnames';
import Item from './ItemUI';
import React from 'react';

import './Items.css';

const ItemsUI = ({items, onToggleIsComplete, onRemoveItem, shouldShowCompleted}) => {
  return items.length > 0 ? (
    <ol className={classnames('items-list')}>
      {items.map(({description, isComplete, uuid}) => {
        return (!shouldShowCompleted && isComplete) ? null : (
          <Item
            description={description}
            isComplete={isComplete}
            key={uuid}
            onChange={onToggleIsComplete}
            onRemove={onRemoveItem}
            uuid={uuid}
          />
        );
      })}
    </ol>
  ) : null;
};

export default ItemsUI;
