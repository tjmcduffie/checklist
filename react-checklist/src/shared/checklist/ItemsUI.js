import classnames from 'classnames';
import ItemUI, {SortableItem} from './ItemUI';
import React from 'react';
import {SortableContainer} from 'react-sortable-hoc';

import './Items.css';

const ItemsUI = ({isSortable, items, onToggleIsComplete, onRemoveItem, shouldShowCompleted}) => {
  const ItemComponent = isSortable ? SortableItem : ItemUI
  return items.length > 0 ? (
    <ol className={classnames('items-list')}>
      {items.map(({description, isComplete, order, uuid}) => {
        return (!shouldShowCompleted && isComplete) ? null : (
          <ItemComponent
            description={description}
            index={order}
            isComplete={isComplete}
            key={uuid}
            onChange={onToggleIsComplete}
            onRemove={onRemoveItem}
            uuid={uuid}
          />
        );
      })}
    </ol>
  ) : <ol />;
};

export default ItemsUI;

const SortableItemsContainer = SortableContainer(
  ({itemsProps}) => (
    <ItemsUI
      isSortable
      {...itemsProps}
    />
  )
);

export const SortableItems = ({onSortEnd, ...itemsProps}) => (
  <SortableItemsContainer
    itemsProps={itemsProps}
    onSortEnd={onSortEnd}
    useDragHandle
    lockAxis="y"
  />
);
