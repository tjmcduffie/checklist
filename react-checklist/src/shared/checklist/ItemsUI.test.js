import ItemsUI from './ItemsUI';
import React from 'react';
import ReactDOM from 'react-dom';
import {create} from 'react-test-renderer';

describe('ItemsUI', () => {
  const noop = () => {};
  const items = [
    {description: 'one', isComplete: false, uuid: 'asdfasdf'},
    {description: 'two', isComplete: true, uuid: 'qwerqwer'},
    {description: 'three', isComplete: false, uuid: 'zxcvzxcv'},
  ];

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ItemsUI items={items} onToggleIsComplete={noop} onRemoveItem={noop} />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  describe('renders predictably', () => {
    it('when showing all items', () => {
      const elem = create(
        <ItemsUI items={items} onToggleIsComplete={noop} onRemoveItem={noop} shouldShowCompleted={true} />
      );
      const tree = elem.toTree();
      expect(tree.rendered.rendered.length).toBe(3);
      expect(tree.rendered.rendered[0].props.uuid).toBe('asdfasdf');
      expect(tree.rendered.rendered[1].props.uuid).toBe('qwerqwer');
      expect(tree.rendered.rendered[2].props.uuid).toBe('zxcvzxcv');
      expect(elem.toJSON()).toMatchSnapshot();
    });

    it('when showing only incomplete items', () => {
      const elem = create(
        <ItemsUI items={items} onToggleIsComplete={noop} onRemoveItem={noop} shouldShowCompleted={false} />
      );
      const tree = elem.toTree();
      expect(tree.rendered.rendered.length).toBe(2);
      expect(tree.rendered.rendered[0].props.uuid).toBe('asdfasdf');
      expect(tree.rendered.rendered[1].props.uuid).toBe('zxcvzxcv');
      expect(elem.toJSON()).toMatchSnapshot();
    });
  });
});
