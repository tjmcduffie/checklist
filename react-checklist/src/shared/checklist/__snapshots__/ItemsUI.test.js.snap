// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsUI renders predictably when showing all items 1`] = `
<ol
  className="items-list"
>
  <li
    className="item flex-row"
  >
    <span
      className="flex-auto item-checkbox-ui"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    />
    <input
      checked={false}
      className="item-checkbox-input"
      hidden={true}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="item-description flex-grow flex-align-center"
    >
      one
    </span>
    <button
      className="clear-button monospaced-button flex-shrink flex-align-top"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      delete
    </button>
  </li>
  <li
    className="item flex-row"
  >
    <span
      className="flex-auto item-checkbox-ui item-checkbox-ui-complete"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    />
    <input
      checked={true}
      className="item-checkbox-input"
      hidden={true}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="item-description-complete flex-grow flex-align-center"
    >
      two
    </span>
    <button
      className="clear-button monospaced-button flex-shrink flex-align-top"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      delete
    </button>
  </li>
  <li
    className="item flex-row"
  >
    <span
      className="flex-auto item-checkbox-ui"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    />
    <input
      checked={false}
      className="item-checkbox-input"
      hidden={true}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="item-description flex-grow flex-align-center"
    >
      three
    </span>
    <button
      className="clear-button monospaced-button flex-shrink flex-align-top"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      delete
    </button>
  </li>
</ol>
`;

exports[`ItemsUI renders predictably when showing only incomplete items 1`] = `
<ol
  className="items-list"
>
  <li
    className="item flex-row"
  >
    <span
      className="flex-auto item-checkbox-ui"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    />
    <input
      checked={false}
      className="item-checkbox-input"
      hidden={true}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="item-description flex-grow flex-align-center"
    >
      one
    </span>
    <button
      className="clear-button monospaced-button flex-shrink flex-align-top"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      delete
    </button>
  </li>
  <li
    className="item flex-row"
  >
    <span
      className="flex-auto item-checkbox-ui"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    />
    <input
      checked={false}
      className="item-checkbox-input"
      hidden={true}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="item-description flex-grow flex-align-center"
    >
      three
    </span>
    <button
      className="clear-button monospaced-button flex-shrink flex-align-top"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      delete
    </button>
  </li>
</ol>
`;
